server/service.go

```diff
@@ -28,24 +33,58 @@ func newService(svcInfo *serviceinfo.ServiceInfo, handler interface{}) *service
 }
 
 type services struct {
-	svcMap     map[string]*service
-	defaultSvc *service
+	svcSearchMap                       map[string]*service // key: "svcName.methodName" and "methodName", value: svcInfo
+	svcMap                             map[string]*service // key: service name, value: svcInfo
+	conflictingMethodHasFallbackSvcMap map[string]bool
+	fallbackSvc                        *service
+	hasCombineService                  bool
 }
 
 func newServices() *services {
-	return &services{svcMap: map[string]*service{}}
+	return &services{
+		svcSearchMap:                       map[string]*service{},
+		svcMap:                             map[string]*service{},
+		conflictingMethodHasFallbackSvcMap: map[string]bool{},
+	}
 }
 
-func (s *services) addService(svcInfo *serviceinfo.ServiceInfo, handler interface{}) {
+func (s *services) addService(svcInfo *serviceinfo.ServiceInfo, handler interface{}, isFallbackService bool) error {
 	svc := newService(svcInfo, handler)
-	if s.defaultSvc == nil {
-		s.defaultSvc = svc
+
+	// combine service check
```

Comment:
The check is a bit obsecure.

Is it that we allow at one CombineService if noother services? If so, the impl may be wrong: one can add a non-combine-service and then add the combine service.

Reply:
Btw, a good comment should be describing "why we need to perform the combine service check".